### PR TITLE
Splitting EXT_mesh_features into multiple extensions

### PR DESCRIPTION
**Pull request by lilleyse**
This PR splits [EXT_mesh_features](https://github.com/CesiumGS/glTF/tree/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features) into three separate extensions:

  * [EXT_mesh_features](https://github.com/CesiumGS/glTF/tree/ext-mesh-features-revision/extensions/2.0/Vendor/EXT_mesh_features) that only defines the concept of feature IDs
  * [EXT_structural_metadata](https://github.com/CesiumGS/glTF/tree/ext-mesh-features-revision/extensions/2.0/Vendor/EXT_structural_metadata) that allows associating the features with metadata
  * [EXT_instance_features](https://github.com/CesiumGS/glTF/tree/ext-mesh-features-revision/extensions/2.0/Vendor/EXT_instance_features) that supports GPU instance feature IDs

This was done for multiple reasons
  * The extension was doing too much
  * Feature identification and metadata can be cleanly decoupled
  * Different metadata encodings may emerge in the future
  * Several users of the extension are interested in feature IDs but want to use their own database

We even considered splitting `EXT_structural_metadata` into four separate extensions - one for the schema definition and three others for the encodings (property tables, property textures, property attributes) - but decided it was better to keep them together.

Opening the PR here for early feedback, then we'll updated the `Khronos/glTF` PRs. This includes the work from https://github.com/CesiumGS/glTF/pull/51 https://github.com/CesiumGS/glTF/pull/52 https://github.com/CesiumGS/glTF/pull/55

#

To do:

* [x] Dictionary vs. array form for `featureIds`
* [ ] From @ptrgags: When the property table offset/scale overrides the parent, should those be checked in lockstep? E.g. what if the class defined `offset` but not `scale` and the property table defines `scale` but not `offset`? even though the property table's offset is undefined, a default (probably 0) would make more sense than inheriting the class offset (which could be different)
* [ ] Allow `propertyAttributes` to refer to components of an accessor. Allows properties to be packed into the same attribute without excessive padding (glTF requires 4-byte padding for attribute elements)

### CODE REVIEW

File: extensions/2.0/Vendor/EXT_instance_features/schema/featureId.schema.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            "minimum": 1,
+            "description": "The number of unique features in the attribute."
+        },
+        "nullFeatureId": {
+            "type": "integer",
+            "minimum": 0,
+            "description": "A value that indicates that no feature is associated with this instance."
+        },
+        "attribute": {
+            "description": "An attribute containing feature IDs. When this is omitted, then the feature IDs are assigned to the GPU instances by their index.",
```
Comment (ptrgags) [inline extensions/2.0/Vendor/EXT_instance_features/schema/featureId.schema.json:24]: @lilleyse is this intentional that GPU instaces can have this implicit range (essentially `constant: 0, repeat: 1`) but primitives _require_ an attribute? (even though the vertex ID could be used)

Comment (lilleyse) [reply extensions/2.0/Vendor/EXT_instance_features/schema/featureId.schema.json:24]: Short term though, I'm good with the vertex ID approach. It's not as expressive as implicit feature ids were before it's easy to understand. Having some implicit feature ID capability is nice, especially once we start storing vector data in glTF. It's much more likely that GIS point features will have string and boolean properties, something that `propertyAttributes` doesn't support.

Comment (lilleyse) [reply extensions/2.0/Vendor/EXT_instance_features/schema/featureId.schema.json:24]: I updated the schema to allow for per-vertex implicit feature IDs. https://github.com/CesiumGS/glTF/pull/56/commits/12f2d6396a346f8fbdb51fb388c81df0cf03b6d6

@javagl could you update the `EXT_mesh_features` text to account for this change?

Comment (javagl) [reply extensions/2.0/Vendor/EXT_instance_features/schema/featureId.schema.json:24]: The PR is at https://github.com/CesiumGS/glTF/pull/57

Comment (lilleyse) [reply extensions/2.0/Vendor/EXT_instance_features/schema/featureId.schema.json:24]: Thanks, merged!

File: extensions/2.0/Vendor/EXT_structural_metadata/README.md [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+>         "properties": {
+>           "species": {
+>             "bufferView": 2,
+>             "stringOffsetBufferView": 3
+>           },
+>           "age": {
+>             "bufferView": 1
+>           },
+>           "height": {
+>             "bufferView": 0
```
[the remark this answers is not in the document]
Comment (lilleyse) [reply extensions/2.0/Vendor/EXT_structural_metadata/README.md]: Yes, fixed in https://github.com/CesiumGS/glTF/pull/56/commits/e023446a953ee4e8d205766cbcc2caffe2afc7cd

**Review by javagl — COMMENTED** [review]
The extensions that this PR consists of had originally been a single extension. The original, single extension has been reviewed. Each PR that carved out one of the new extensions has been created in a dedicated PR that went into the branch that this PR is based on. There _have_ been changes in each individual part, but these should be consistent with the overall structure and description. So I think this could be merged.

Only one question: There are some links like `https://github.com/CesiumGS/3d-tiles/blob/extension-revisions/next/REVISION_HISTORY.md` that refer to a branch (and not `main`) - at which point are these supposed to be be updated?

**Pull request merged**